contributing/naming-conventions.md

```diff
@@ -0,0 +1,150 @@
+
+
+## Components
+
+Use `PascalCase` for component names.
+
+```diff
+-<Menubutton />
++<MenuButton />
+```
+
+Prefer simple component names. Namespacing is unnecessary.
+
+```diff
+-<FdsButton />
++<Button />
+```
+
+Prefer props for declaring variants instead of separate components.
+
+```diff
+-<PrimaryButton />
+-<SecondaryButton />
++<Button type="primary|secondary" />
+```
+
+When there is a parent-child relationship between two components, make that relationship
+explicit through naming:
+
+```diff
+-<Item /> /* 💭 what kind of item? */
++<FlexItem /> /* 💭 this looks like a child of a `Flex`! */
+```
+
+## Props
+
+Props for event handlers should include an `on` prefix:
+
+```diff
+-hover
++onHover
+```
+
+Boolean props should have an auxiliary verb prefix:
+
+```diff
+-selected
++isSelected
+
+-closeButton
++hasCloseButton
+
+-portal
++usePortal
+```
+
+When a prop expects a React component instead of a primitive, use `PascalCase`:
+
+```diff
+-icon
++Icon
+```
+
+Props that accept a list of items should always be plural:
+
+```diff
+-list={['thing one', 'thing two']}
++listItems={['thing one', 'thing two']}
+```
+
+## CSS classes
+In FDS, we follow a modified BEM naming convention. With our convention, hyphens carry the
+meaning for block-element relationships and underscores are reserved for conditional
+classes.
+
+```css
+/* parent-child */
+.base-child {}
+
+/* modifier */
+.base--modifier {}
+
+/* child modifier */
+.base-child--modifier {}
+
+/* conditional (not in use yet) */
+.condition_base--modifier {}
+```
+
+#### Base
+The word at the beginning of a class name is the "base" of the class. This could
+refer to an element, a CSS property, or a concept as a standalone entity.
+
+CSS properties as "base":
+
+- **`.margin`**`--top`
+- **`.fontSize`**`--xl`
+- **`.color`**`--red`
+
+Concepts that do not map to a single CSS property can also be a "base" name:
+
+- **`.flush`**`--all`
+- **`.childAlign`**`--center--center`
+- **`.media`**`--xl`
+
+
+#### Child
+Parent-child relationships are denoted by a single hyphen. This allows us to keep our
+selectors flat (a single class without ancestry) while providing a hint that the selector
+is in some way related to a parent element.
+
+```css
+.flex-flexItem {} /* the flexItem is a child of "flex" */
+.button-icon {} /* this is for an icon element that is a child of a "button" */
+```
+
+#### Modifier
+A double hyphen indicates the class modifies some style property.
+
+- `.flush`**`--all`**
+- `.media`**`--xl`**
+- `.menu-item`**`--selected`**
+
+#### Conditional
+Some classes may apply their styles conditionally based on browser support, viewport size,
+or other contextual condition. We indicate this with a prefix and underscore at the
+beginning of the class name.
+
+- **`.atMedium_`**`margin--all` (adds margin to all sides when the viewport reaches the
+- medium breakpoint)
+- **`.atLarge_`**`display--block` (sets display to block when the viewport reaches the
+- large breakpoint)
```

Comment:
I know you marked conditional classes as not existing yet above... it may be worth not calling out these utilities until they exist... I can totally see some developer trying to use this.